packages/react-instantsearch-core/src/widgets/DynamicWidgets.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { Fragment, ReactChild, ReactNode } from 'react';
+import { getDisplayName } from '../core/utils';
+import connectDynamicWidgets from '../connectors/connectDynamicWidgets';
+
+function getAttribute(component: ReactChild): string | undefined {
+  if (typeof component !== 'object') {
+    return undefined;
+  }
+
+  if (component.props.attribute) {
+    return component.props.attribute;
+  }
+  if (Array.isArray(component.props.attributes)) {
+    return component.props.attributes[0];
+  }
```

Comment:
I'm thinking about cases where perhaps using the first attribute could be limiting.

If we process the component if _any_ attributes match the `attributes` prop, it might become more flexible for users—especially in the context of the "scoped hierarchical menu" à la Amazon where the root changes (or is the `root` prop enough in that use case?).

Reply:
I assumed for hierarchical menus that people would set the first/root attribute as the one one to render hierarchically. If the menu is only displayed from a certain level onwards, that probably would be done either with a refinement and ignoring the levels that aren't shown or using `rootPath` for this. Concatenating or an array of attributes seems wrong, since for example in current refinement we use rootPath too